lib/transaction.js

```diff
@@ -81,12 +98,16 @@ Transaction.prototype.commit = function() {
     .sequelize
     .getQueryInterface()
     .commitTransaction(this, this.options)
-    .finally(function() {
+    .finally(function () {
       self.finished = 'commit';
       if (!self.options.transaction) {
         self.cleanup();
       }
-    });
+    })
+    .tap(
+      function (val) { self.promise.seqResolve(val); },
+      function (err) { self.promise.seqReject(err); }
```

Comment:
You're right.
So it means that we can access directly into `promise._resolveCallback` and `promise._rejectCallback` and use them here, instead of getting them from resolver execution (it's too nasty, thats why its deprecated aswell).